routing/router.go

```diff
@@ -616,373 +450,156 @@ func (r *ChannelRouter) networkHandler() {
 		}
 
-		if err = r.cfg.Graph.AddLightningNode(node); err != nil {
-			log.Errorf("unable to add node %v: %v", msg.NodeID, err)
-			return false
-		}
-
-		log.Infof("Updated vertex data for node=%x",
-			msg.NodeID.SerializeCompressed())
+		log.Infof("Updated vertex data for msg=%x",
+			msg.PubKey.SerializeCompressed())
 
-	// A new channel announcement has arrived, this indicates the
-	// *creation* of a new channel within the graph. This only advertises
-	// the existence of a channel and not yet the routing policies in
-	// either direction of the channel.
-	case *lnwire.ChannelAnnouncement:
+	case *channeldb.ChannelEdgeInfo:
 		// Prior to processing the announcement we first check if we
 		// already know of this channel, if so, then we can exit early.
-		channelID := msg.ChannelID.ToUint64()
-		_, _, exists, err := r.cfg.Graph.HasChannelEdge(channelID)
+		_, _, exists, err := r.cfg.Graph.HasChannelEdge(msg.ChannelID)
 		if err != nil && err != channeldb.ErrGraphNoEdgesFound {
-			log.Errorf("unable to check for edge existence: %v", err)
-			return false
+			return errors.Errorf("unable to check for msg "+
+				"existence: %v", err)
+
 		} else if exists {
-			log.Debugf("Ignoring announcement for known chan_id=%v",
-				channelID)
-			return false
+			return errors.Errorf("ignoring msg for known "+
+				"chan_id=%v", msg.ChannelID)
 		}
 
-		// If the advertised inclusionary block is beyond our knowledge
-		// of the chain tip, then we'll put the announcement in limbo
-		// to be fully verified once we advance forward in the chain.
-		if isPremature(&msg.ChannelID) {
-			blockHeight := msg.ChannelID.BlockHeight
-			log.Infof("Announcement for chan_id=(%v), is "+
-				"premature: advertises height %v, only height "+
-				"%v is known", channelID,
-				msg.ChannelID.BlockHeight, r.bestHeight)
-
-			r.prematureAnnouncements[blockHeight] = append(
-				r.prematureAnnouncements[blockHeight],
-				msg,
-			)
-			return false
-		}
+		// TODO(andrew.shvv) Add validation that bitcoin keys are
+		// binded to the funding transaction.
 
 		// Before we can add the channel to the channel graph, we need
 		// to obtain the full funding outpoint that's encoded within
 		// the channel ID.
-		fundingPoint, err := r.fetchChanPoint(&msg.ChannelID)
+		channelID := lnwire.NewChanIDFromInt(msg.ChannelID)
+		fundingPoint, err := r.fetchChanPoint(&channelID)
 		if err != nil {
-			log.Errorf("unable to fetch chan point for chan_id=%v: %v",
-				channelID, err)
-			return false
+			return errors.Errorf("unable to fetch chan point for "+
+				"chan_id=%v: %v", msg.ChannelID, err)
 		}
 
 		// Now that we have the funding outpoint of the channel, ensure
 		// that it hasn't yet been spent. If so, then this channel has
 		// been closed so we'll ignore it.
-		chanUtxo, err := r.cfg.Chain.GetUtxo(&fundingPoint.Hash,
-			fundingPoint.Index)
+		chanUtxo, err := r.cfg.Chain.GetUtxo(&fundingPoint.Hash, fundingPoint.Index)
```

Comment:
The discovery package should tasked fetching the full details required to insert the new edge into the database. With this change, the router assume all information it receives has been fully validated.

So fetching the outpoint, and also the utxo should be moved into the discovery package.